app/src/main/java/app/envelop/domain/LoginService.kt

```diff
@@ -42,15 +42,17 @@ class LoginService
 
     return rxSingle {
       val userData = blockstackProvider.get().handlePendingSignIn(authResponseTokens.last())
-      if (userData.hasValue && userData.value.isComplete()) {
+      if (userData.hasValue && userData.value.containsValidUsername()) {
         userRepository.setUser(userData.value?.toUser())
+      } else if (!userData.value.containsValidUsername()) {
+        throw UnverifiedUsername("Invalid Username")
       } else {
         throw Error(userData.error?.message)
       }
     }.toOperation()
   }
 
-  private fun UserData?.isComplete() =
+  private fun UserData?.containsValidUsername() =
     this?.let {
       val username = it.json.optString("username")
       (!username.isNullOrBlank() && username != "null")
```

Comment:
Também poderia ser:
```
this?.json?.optString("username")?.let { username ->
      (!username.isNullOrBlank() && username != "null")
} ?: false
```
Mas não sei se fica mais claro...

Reply:
Eu diria que sim 😄

Reply:
mas não há necessidade de checkar null então:

`  private fun UserData?.isComplete() =
    this?.json?.optString("username")?.let {username ->
      (!username.isBlank() && username != "null")
    } ?: false`